### PR TITLE
feat(products): add API to fetch all products

### LINKED ISSUE #700: api for fetching products data
### Description

products displaying on shop page is static , API for fetching data from database and displaying it dynamically would be better

### Screenshots

_No response_

### 🥦 Browser

Google Chrome

### Checklist ✅

- [X] I checked and didn't find similar issue
- [X] I have read the [Contributing Guidelines](https://github.com/IAmTamal/Milan/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (blank for no)

### PR DESCRIPTION
**Pull request by Nitish-hack**
### Related Issue 

Closes: #700 

#### Changes made 👷🏻‍♂️

feat:  add API to fetch all products
i have created an API to fetch all products from the database , i have made changes in product routes

#### Screenshots  📸

image
image

### CODE REVIEW

**Comment by Nitish-hack** [comment]
any update?

File: server/routes/Products.js [hunk 1]
```diff
@@ -36,4 +36,22 @@ router.post("/addproduct", async (req, res) => {
...
+/**
+ * @description Get all products
+ * @route GET /products
+ * @access Public
+ * @returns Array of all products
+ */
+router.get("/allproducts", async (req, res) => {
+  try {
+    const allProducts = await Products.find();
+    res.json(allProducts);
```
Comment (tamalCodes) [inline server/routes/Products.js:50]: Can we also add a status code for the successful response  ?  Maybe like `res.status(200)` ?

**Review by tamalCodes — APPROVED** [review]
The rest all looks good to me !

File: server/routes/Products.js (hunk 1 above)
Comment (Nitish-hack) [reply server/routes/Products.js]: yeah sure , done!

**Review by tamalCodes — APPROVED** [review]
**Amazing work,**
This looks good to me, will be merged soon 🎉.

**Comment by Nitish-hack** [comment]
thanks

**Pull request merged**